jerry-core/parser/js/js-parser-module.c

```diff
@@ -173,393 +199,275 @@ parser_module_add_export_node_to_context (parser_context_t *context_p) /**< pars
     }
-    else
+
+    if (parser_module_check_duplicate_import (context_p, local_name_p))
     {
-      parser_raise_error (context_p, PARSER_ERR_RIGHT_PAREN_COMMA_FROM_EXPECTED);
+      ecma_deref_ecma_string (local_name_p);
+      ecma_deref_ecma_string (import_name_p);
+      parser_raise_error (context_p, PARSER_ERR_DUPLICATED_IMPORT_BINDING);
     }
-    lexer_next_token (context_p);
-  }
-} /* parser_module_parse_import_item_list */
 
-/**
- * Handle import requests.
- * Check if imported variables are exported in the appropriate module.
- * Raise parser error if imported item is not exported.
- */
-void
-parser_module_handle_requests (parser_context_t *context_p) /**< parser context */
-{
-  parser_module_context_t *module_context_p = JERRY_CONTEXT (module_top_context_p);
-
-  if (context_p->module_context_p->exports_p == NULL || module_context_p == NULL)
-  {
-    return;
-  }
-
-  bool throw_error = false;
-
-  parser_module_names_t *import_name_p = module_context_p->imports_p->module_names_p;
-  parser_module_names_t *current_exports_p = context_p->module_context_p->exports_p->module_names_p;
+    parser_module_add_names_to_node (context_p, import_name_p, local_name_p);
+    ecma_deref_ecma_string (local_name_p);
+    ecma_deref_ecma_string (import_name_p);
 
-  for (uint16_t i = 0; i < module_context_p->imports_p->module_request_count; ++i)
-  {
-    bool request_is_found_in_module = false;
-    parser_module_names_t *export_iterator_p = current_exports_p;
-
-    for (uint16_t j = 0; j < context_p->module_context_p->exports_p->module_request_count;
-         ++j, export_iterator_p = export_iterator_p->next_p)
+    if (context_p->token.type != LEXER_COMMA
+        && (context_p->token.type != LEXER_RIGHT_BRACE))
     {
-      if (ecma_compare_ecma_strings (import_name_p->local_name_p, export_iterator_p->import_name_p))
-      {
-        request_is_found_in_module = true;
-        break;
-      }
+      parser_raise_error (context_p, PARSER_ERR_RIGHT_BRACE_COMMA_EXPECTED);
     }
-
-    if (!request_is_found_in_module)
+    else if (context_p->token.type == LEXER_COMMA)
     {
-      parser_module_free_saved_names (context_p->module_context_p->exports_p);
-      throw_error = true;
-      break;
+      lexer_next_token (context_p);
     }
 
-    import_name_p = import_name_p->next_p;
-  }
-
-  *module_context_p->exports_p = *context_p->module_context_p->exports_p;
-  parser_free (context_p->module_context_p->exports_p, sizeof (parser_module_node_t));
-
-  if (throw_error)
-  {
-    parser_raise_error (context_p, PARSER_ERR_MODULE_REQUEST_NOT_FOUND);
+    if (context_p->token.type == LEXER_LITERAL
+        && lexer_compare_raw_identifier_to_current (context_p, "from", 4))
+    {
+      parser_raise_error (context_p, PARSER_ERR_RIGHT_BRACE_EXPECTED);
+    }
   }
-} /* parser_module_handle_requests */
+} /* parser_module_parse_import_clause */
 
 /**
  * Raises parser error if the import or export statement is not in the global scope.
```

Comment:
The builtin `Function(args, src)` also can trigger the parser. Is this handled below?